pkg/autoscaler/stats_scraper.go

```diff
@@ -104,23 +104,23 @@ func newServiceScraperWithClient(
 		return nil, fmt.Errorf("no Revision label found for Metric %s", metric.Name)
 	}
 
-	serviceName := reconciler.GetServingK8SServiceNameForObj(revName)
+	serviceName := reconciler.GetMetricsK8SServiceNameForObj(revName)
 	return &ServiceScraper{
-		httpClient:      httpClient,
-		endpointsLister: endpointsInformer.Lister(),
-		url:             fmt.Sprintf("http://%s.%s:%d/metrics", serviceName, metric.Namespace, v1alpha1.RequestQueueMetricsPort),
-		metricKey:       NewMetricKey(metric.Namespace, metric.Name),
-		namespace:       metric.Namespace,
-		revisionService: reconciler.GetServingK8SServiceNameForObj(revName),
+		httpClient:          httpClient,
+		endpointsLister:     endpointsInformer.Lister(),
+		url:                 fmt.Sprintf("http://%s.%s:%d/metrics", serviceName, metric.Namespace, v1alpha1.RequestQueueMetricsPort),
+		metricKey:           NewMetricKey(metric.Namespace, metric.Name),
+		namespace:           metric.Namespace,
+		scrapeTargetService: reconciler.GetServingK8SServiceNameForObj(revName),
```

Comment:
This should just be `serviceName` as well. No need to check on both services I think.

Reply:
Done.